core/sdk-core/src/test/java/software/amazon/awssdk/core/RequestOverrideConfigurationTest.java

```diff
@@ -27,16 +26,53 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import nl.jqno.equalsverifier.EqualsVerifier;
 import org.junit.Test;
 import software.amazon.awssdk.core.interceptor.ExecutionAttribute;
 import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.signer.Signer;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
 import software.amazon.awssdk.metrics.MetricPublisher;
 import software.amazon.awssdk.utils.ImmutableMap;
 
 public class RequestOverrideConfigurationTest {
     private static final String HEADER = "header";
     private static final String QUERY_PARAM = "queryparam";
 
+    @Test
+    public void equalsHashcode() {
+        EqualsVerifier.forClass(RequestOverrideConfiguration.class)
+                      .usingGetClass()
+                      .verify();
+    }
+
+    @Test
+    public void toBuilder_minimal() {
+        RequestOverrideConfiguration configuration = SdkRequestOverrideConfiguration.builder()
+                                                                                    .build();
+
+        assertThat(configuration.toBuilder().build()).usingRecursiveComparison().isEqualTo(configuration);
+    }
+
+    @Test
+    public void toBuilder_maximal() {
+        ExecutionAttribute testAttribute = new ExecutionAttribute("TestAttribute");
+        String expectedValue = "Value1";
+
+        RequestOverrideConfiguration configuration = SdkRequestOverrideConfiguration.builder()
+                .putHeader(HEADER, "foo")
+                .putRawQueryParameter(QUERY_PARAM, "foo")
+                .addApiName(a -> a.name("test1").version("1"))
+                .apiCallTimeout(Duration.ofSeconds(1))
+                .apiCallAttemptTimeout(Duration.ofSeconds(1))
+                .signer(new NoOpSigner())
+                .executionAttributes(ExecutionAttributes.builder().put(testAttribute, expectedValue).build())
+                .addMetricPublisher(mock(MetricPublisher.class))
+                .build();
+
+        assertThat(configuration.toBuilder().build()).usingRecursiveComparison().isEqualTo(configuration);
```

Comment:
It would be nice if we could find a way to enforce this validation for all of our classes that implement `toBuilder()`.